### DISCUSSION TITLE
E2 Entity Ownership Transfer

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2659] [Suggestions]
E2 function that allows *entity* ownership to transfer to another player. Probably dictated by a server convar (none, everyone, friends, admin-only) and another for the client (allow/disallow).

Example function signature `entity:transferOwner(entity player)`.

**Purpose:**
I know some people like making complex games with E2 and there are certain actions you can't (easily) do with certain entities on other players. This would allow users to accomplish those actions simply by having those options transfer that entity to the target player. Convars, CPPI, and explicit user approval could get rid of most of the abuse potential.

**Considerations:**
Should E2s be allowed to be transferred?

**Comment by thegrb93** [comment]
Too complicated

**Comment by Fasteroid** [comment]
I see no reason to implement this, BUT, I would completely support a native e2 function to share an entity through CPPI.
